src/JustSaying/Messaging/Channels/SubscriptionGroups/SubscriptionConfigBuilder.cs

```diff
@@ -0,0 +1,130 @@
+        /// Defaults to 100
+        /// </summary>
+        /// <param name="multiplexerCapacity">The maximum multiplexer capacity</param>
+        /// <returns>This builder object.</returns>
+        public SubscriptionConfigBuilder WithDefaultMultiplexerCapacity(int multiplexerCapacity)
+        {
+            DefaultMultiplexerCapacity = multiplexerCapacity;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the default number of messages to try and fetch from SQS per attempt for each queue in a <see cref="ISubscriptionGroup"/>
+        /// Defaults to 10
+        /// </summary>
+        /// <param name="prefetch">the number of messages to load per request</param>
+        /// <returns>This builder object.</returns>
+        public SubscriptionConfigBuilder WithDefaultPrefetch(int prefetch)
+        {
+            DefaultPrefetch = prefetch;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the default maximum number of messages that may be processed at once by a <see cref="ISubscriptionGroup"/>.
+        /// Defaults to Environment.ProcessorCount * 4
+        /// </summary>
+        /// <param name="concurrencyLimit">The maximum number of messages to process at the same time</param>
+        /// <returns>This builder object.</returns>
+        public SubscriptionConfigBuilder WithDefaultConcurrencyLimit(int concurrencyLimit)
+        {
+            DefaultConcurrencyLimit = concurrencyLimit;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the default number of messages that will be buffered from SQS for each of the queues in a <see cref="ISubscriptionGroup"/>
+        /// before waiting for them to drain into the <see cref="IMultiplexer"/>.
+        /// Note: This setting is per-queue. To set the shared buffer size for all queues, see <see cref="WithDefaultMultiplexerCapacity"/>
+        /// </summary>
+        /// <param name="bufferSize">The maximum number of messages for each queue to buffer</param>
+        /// <returns>This builder object.</returns>
+        public SubscriptionConfigBuilder WithDefaultBufferSize(int bufferSize)
+        {
+            DefaultBufferSize = bufferSize;
+            return this;
+        }
+
+        /// <summary>
+        /// Overrides the default middleware used by the receive pipeline, which performs some default error handling
+        /// (see <see cref="DefaultSqsMiddleware"/>)
+        /// </summary>
+        /// <param name="middleware">A <see cref="ReceiveMiddleware"/> that replaces the default middleware
+        /// (see <see cref="DefaultSqsMiddleware"/>)</param>
+        /// <returns>The builder object</returns>
+        public SubscriptionConfigBuilder WithCustomMiddleware(ReceiveMiddleware middleware)
+        {
+            SqsMiddleware = middleware;
+            return this;
+        }
+
+        public void Validate()
+        {
+            if (DefaultPrefetch < 0)
```

Comment:
I think the extra mechanics of collecting errors and throwing a meta-exception at the end pays off when there's a desire to show all errors at once like on a UI, but in this case I'd argue that we want to keep it as simple as possible and fail fast. 

Given that most of these settings have sane defaults, it's only when overriding them to nonsense values that devs would hit these validations - like setting a negative pre-fetch. Do you think it would be worthwhile having the extra machinery for cases where the dev has made multiple mistakes?

Reply:
> Do you think it would be worthwhile having the extra machinery for cases where the dev has made multiple mistakes?

Yes. Only because I have been on the receiving end of this type of thing before (when using some other 3rd party library). I got very frustrated by having to iterate through all the errors one by one.

Reply:
Similar to the previous comment, if we want to overhaul all the validation, that should be a separate piece of work.